spec/stack.js

```diff
@@ -19,4 +19,37 @@ describe('Stack', () => {
         .to.alter(() => myStack.length(), { from: 0, to: 1 })
     })
   })
+
+  context('pop()', () => {
+    it('returns and removes the top element in a stack or null if stack is empty', () => {
```

Comment:
When I see words like `and` and `or` in a test description, I suspect that the test is trying to test too many things simultaneously.  Consider splitting this into two tests:
1. A test that the element is removed (i.e. that the size of the stack changes)
1. A test that the correct element is returned